File: Day101_2025/Q101_Leetcode.java
class Q101_Leetcode {
    public int countSymmetricIntegers(int low, int high) {
        int res = 0;

        for (int num = low; num <= high; num++) {
            String s = Integer.toString(num);
            int len = s.length();
            if (len % 2 != 0) continue; // skip odd-digit numbers

            int half = len / 2;
            int leftSum = 0, rightSum = 0;

            for (int i = 0; i < half; i++) {
                leftSum += s.charAt(i) - '0';
                rightSum += s.charAt(i + half) - '0';
            }

            if (leftSum == rightSum) {
                res++;
            }
        }

        return res;
    }
}

File: Day101_2025/Q101_Leetcode.py
class Solution:
    def countSymmetricIntegers(self, low: int, high: int) -> int:
        res = 0

        for num in range(low, high + 1):
            s = str(num)
            if len(s) % 2 != 0:
                continue

            half = len(s) // 2
            left = sum(int(d) for d in s[:half])
            right = sum(int(d) for d in s[half:])

            if left == right:
                res += 1

        return res
